Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v1/WallpaperSettingsViewController.swift

```diff
@@ -266,6 +242,24 @@ private extension WallpaperSettingsViewController {
         // Reload the complete collection view as the section headers are not adjusting their size correctly otherwise
         collectionView.reloadData()
     }
+
+    func downloadAndSetWallpaper(at indexPath: IndexPath) {
+        activityIndicatorView.startAnimating()
+        viewModel.downloadAndSetWallpaper(at: indexPath) { [weak self] result in
+            ensureMainThread {
+                switch result {
+                case .success:
+                    self?.showToast()
+                case .failure(let error):
+                    self?.browserLog.info(error.localizedDescription)
+                    self?.showError(error) { _ in
+                        self?.downloadAndSetWallpaper(at: indexPath)
+                    }
```

Comment:
Just curious for showing an error we are downloading wallpaper again 🤔

Reply:
The alert controller that is shown has two buttons: Cancel and Retry. The closure you are seeing here is for retry and is named `retryHandler` in the method.